prov/util/src/util_attr.c

```diff
@@ -56,11 +71,148 @@ static int fi_valid_addr_format(uint32_t prov_format, uint32_t user_format)
 	}
 }
 
+/*
+ * Accepts a string which has tokens delimited by underscores and returns a token
+ * array, tok[0..max_token - 1]. Callers are responsible for freeing tok[0] which
+ * will point to a fudged copy of the string. Refer strtok_r manpage for details.
+ */
+int utilx_parse_name(const char *name, int max_tok, char **tok, int get_prefix)
+{
+	char *name_dup, *name_ref, *save_ptr;
+	char *delim = "_";
+	int i, parsed_len = 0;
+
+	name_dup = strdup(name);
+	if (!name_dup)
+		return -FI_ENOMEM;
+
+	name_ref = name_dup;
+	for (i = 0; i < max_tok; name_ref = NULL, i++) {
+		tok[i] = strtok_r(name_ref, delim, &save_ptr);
+		if (!tok[i])
+			break;
+	}
+	if (i < max_tok)
+		goto err;
+
+	if (!get_prefix) {
+		/* Handle the case where there is a delim in token itself */
+		for (i = 0; i < max_tok; i++)
+			parsed_len += strlen(tok[i]) + 1;
+		parsed_len--;
+
+		if (strlen(name) > parsed_len)
+			tok[max_tok - 1][strlen(tok[max_tok - 1])] = *delim;
+	}
+
+	return 0;
+err:
+	free(name_dup);
+	return -FI_EOTHER;
+}
```

Comment:
There's something about adding a name parsing function that rubs me the wrong way.  It seems like overkill and assumes some sort of format to a fabric name that is not necessarily true.  Why isn't a prefix check on a name sufficient?